nf_core/lint/schema_lint.py

```diff
@@ -2,6 +2,22 @@
 
 import logging
 import nf_core.schema
+import jsonschema
+
+
+def remove_required_fields(schema):
+    """ Remove all required fields from a schema """
+    for group_key in schema["definitions"].keys():
+        group = schema["definitions"][group_key]
+        group_keys = list(group.keys())
+        if "required" in group_keys:
+            required_params = group["required"]
+            for rp in required_params:
+                schema["definitions"][group_key]["properties"].pop(rp)
+            # remove the 'required' key as well
+            schema["definitions"][group_key].pop("required")
```

Comment:
I think that this whole block can probably be a fair bit simpler.. Off the top of my head:

```suggestion
    for group_key in schema["definitions"]:
        try:
            del schema["definitions"][group_key]
        except KeyError as e:
            pass
```

Reply:
wouldn't that just delete all the groups from the schema?

Reply:
Maybe.. Getting a bit lost in this diff now 😅